### PR TITLE
[mysql] Fix the fetchSize parameter is not used

### PR DESCRIPTION
**Pull request by fuyun2024**
issues : #1582

### CODE REVIEW

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java [hunk 1]
```diff
@@ -270,7 +270,7 @@ public MySqlSourceConfig createConfig(int subtaskId) {
         props.setProperty("database.user", checkNotNull(username));
         props.setProperty("database.password", checkNotNull(password));
         props.setProperty("database.port", String.valueOf(port));
-        props.setProperty("database.fetchSize", String.valueOf(fetchSize));
```
Comment (ruanhang1993) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java:273]:
```java
props.setProperty(QUERY_FETCH_SIZE.name(), String.valueOf(fetchSize));
if (fetchSize > 0) {
    // https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html
    jdbcProperties.setProperty("useCursorFetch", "true");
}
```

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java [hunk 2]
```diff
@@ -317,6 +317,8 @@ public MySqlSourceConfig createConfig(int subtaskId) {
         if (jdbcProperties == null) {
             jdbcProperties = new Properties();
         }
+        // https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html
+        jdbcProperties.setProperty("useCursorFetch", "true");
```
Comment (ruanhang1993) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java:321]: Move this line near its relative code.

**Review by ruanhang1993 — COMMENTED** [review]
Thanks for fixing this bug, @fuyun2024

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java (hunk 2 above)
Comment (fuyun2024) [reply flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java]: Hi @ruanhang1993  Thank you for your advice, but 'jdbcProperties' is initialized here. Do I need to move it here?

Comment (ruanhang1993) [reply flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java]: I think we could initialize the `jdbcProperties` earlier.

Comment (fuyun2024) [reply flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java]: hi @ruanhang1993 , I think the important code is at the top, but it is not the most important, so I put it at the end

**Comment by GOODBOY008** [comment]
@fuyun2024  `MySqlTimezoneITCase` hangs in test.

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java [hunk 3]
```diff
@@ -318,6 +318,13 @@ public MySqlSourceConfig createConfig(int subtaskId) {
             jdbcProperties = new Properties();
         }
 
+        // use cursor-based streaming to retrieve a set number of rows
+        // https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-implementation-notes.html
+        if (fetchSize > 0) {
+            jdbcProperties.setProperty("useCursorFetch", "true");
+            props.setProperty(QUERY_FETCH_SIZE.name(), String.valueOf(fetchSize));
```
Comment (GOODBOY008) [inline flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java:325]: The `QUERY_FETCH_SIZE` property is useless for mysql.

**Review by GOODBOY008 — COMMENTED** [review]
@fuyun2024  It‘s not easy to support `useCursorFetch`.

File: flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java (hunk 3 above)
Comment (fuyun2024) [reply flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/config/MySqlSourceConfigFactory.java:325]: Hi @GOODBOY008  I use the mysql 8.0.21 driver and it works

**Comment by ruanhang1993** [comment]
@fuyun2024 Maybe we need to add some tests to verify the bug is fixed.

**Comment by ruanhang1993** [comment]
If we open the `useCursorFetch`, the tests in the  `MySqlTimezoneITCase` will fail.
It seems that the date type field can not be parsed right when `useCursorFetch` is true.
截屏2022-10-31 下午7 28 17

**Comment by ruanhang1993** [comment]
> If we open the `useCursorFetch`, the tests in the `MySqlTimezoneITCase` will fail. It seems that the date type field can not be parsed right when `useCursorFetch` is true. 截屏2022-10-31 下午7 28 17

This problem has been fixed in DBZ-3238(https://github.com/debezium/debezium/pull/2245).
So I have applied these changes to mysql cdc connector.

**Comment by fuyun2024** [comment]
HI @ruanhang1993 , Thank you for your time and suggestions,  @GOODBOY008  has submitted a new PR  #1629 , and I will close this submission later.

**Pull request closed without merging**